### PR TITLE
lock whisper files during carbon-sync and whisper-fill

### PR DESCRIPTION
**Pull request by filippog**
as discussed on #47, the option isn't symmetric between sync and fill (config file vs command line) however fill doesn't take a config file anyway. Also I noticed whisper-fill is included in graphite's distribution too so that will need updating as well.

### CODE REVIEW

**Comment by filippog** [comment]
also note this is missing some prose about the new option and its implication when syncing running/live clusters, I can take a stab at that but I'm no native speakr, @jssjr you'd do a much better job I guess :)

**Comment by filippog** [comment]
updated to reflect latest `master`, thoughts?

**Comment by filippog** [comment]
@jssjr thoughts on this? I think it'd be important to have and avoid multiple writers corrupt whisper files

File: carbonate/fill.py [hunk 1]
```diff
@@ -85,8 +85,13 @@ def fill(src, dst, tstart, tstop):
             return
 
 
-def fill_archives(src, dst, startFrom):
-    header = info(dst)
+def fill_archives(src, dst, startFrom, lock):
```
Comment (deniszh) [inline carbonate/fill.py]: Maybe `def fill_archives(src, dst, startFrom, lock=False)` is better? No need to fix tests then.

**Comment by deniszh** [comment]
Nah, maybe I nitpicking.

**Comment by deniszh** [comment]
Looks like more test need to be fixed

**Comment by deniszh** [comment]
Fixed in https://github.com/graphite-project/carbonate/pull/71

**Pull request closed without merging**